<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="RVWidget" name="widgetRV">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>300</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Utiliser la souris pour bouger le cube</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="animationButton">
          <property name="toolTip">
           <string notr="true">Activer l'animation</string>
          </property>
          <property name="text">
           <string>Animation</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Réglages caméra</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Ouverture focale</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDial" name="fov">
             <property name="toolTip">
              <string>Change la focale de la caméra</string>
             </property>
             <property name="minimum">
              <number>2</number>
             </property>
             <property name="maximum">
              <number>120</number>
             </property>
             <property name="pageStep">
              <number>5</number>
             </property>
             <property name="value">
              <number>45</number>
             </property>
             <property name="tracking">
              <bool>true</bool>
             </property>
             <property name="notchTarget">
              <double>5.000000000000000</double>
             </property>
             <property name="notchesVisible">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QDoubleSpinBox" name="x_light">
            <property name="value">
             <double>10.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="y_light"/>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="z_light">
            <property name="value">
             <double>10.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="ambientButton">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(128, 128, 128);
border: 1px solid black;
border-radius: 5px;</string>
          </property>
          <property name="text">
           <string>Couleur ambiante</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="diffuseButton">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);
border: 1px solid black;
border-radius: 5px;</string>
          </property>
          <property name="text">
           <string>Couleur diffuse</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="specularButton">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);
border: 1px solid black;
border-radius: 5px;</string>
          </property>
          <property name="text">
           <string>Couleur spéculaire</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Coéfficient de refléction spéculaire</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="coeff_spec">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RVWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>rvwidget.h</header>
   <slots>
    <signal>signal1()</signal>
    <slot>startAnimation()</slot>
    <slot>changeFov(int)</slot>
    <slot>changeOpacity(int)</slot>
    <slot>changeWireFrame(bool)</slot>
    <slot>changeCulling(bool)</slot>
    <slot>changeScale(int)</slot>
    <slot>changeSaturation(int)</slot>
    <slot>changeCameraType(bool)</slot>
    <slot>changeSpecCoeff(int)</slot>
    <slot>changeLightX(double)</slot>
    <slot>changeLightY(double)</slot>
    <slot>changeLightZ(double)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>animationButton</sender>
   <signal>clicked()</signal>
   <receiver>widgetRV</receiver>
   <slot>startAnimation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>708</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fov</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeFov(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>742</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>coeff_spec</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeSpecCoeff(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>639</x>
     <y>396</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>395</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x_light</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeLightX(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>598</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>481</x>
     <y>293</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>y_light</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeLightY(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>657</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>486</x>
     <y>340</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>z_light</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widgetRV</receiver>
   <slot>changeLightZ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>728</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>366</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>diffuseButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeDiffuseLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>704</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>857</x>
     <y>320</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>specularButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeSpecularLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>774</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>878</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ambientButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeAmbientLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>777</x>
     <y>297</y>
    </hint>
    <hint type="destinationlabel">
     <x>893</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changeAmbientLight()</slot>
  <slot>changeDiffuseLight()</slot>
  <slot>changeSpecularLight()</slot>
 </slots>
</ui>
